homeassistant/components/huawei_smart_logger/sensor.py

```diff
@@ -0,0 +1,257 @@
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    HuaweiSmartLogger3000EntityDescription(
+        key="currentday_discharge_capacity",
+        translation_key="currentday_discharge_capacity",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    HuaweiSmartLogger3000EntityDescription(
+        key="total_charge",
+        translation_key="total_charge",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    HuaweiSmartLogger3000EntityDescription(
+        key="total_discharge",
+        translation_key="total_discharge",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    HuaweiSmartLogger3000EntityDescription(
+        key="rated_ess_power",
+        translation_key="rated_ess_power",
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Huawei Smart Loggers sensors."""
+    coordinator = hass.data[DOMAIN]
+    _LOGGER.debug("In sensor.py async_setup_entry")
+    async_add_entities(
+        HuaweiSmartLogger3000Sensor(coordinator, description)
+        for description in SENSOR_TYPES
+    )
+
+
+class HuaweiSmartLogger3000Sensor(
+    CoordinatorEntity[HuaweiSmartLogger3000DataCoordinator], SensorEntity
+):
+    """Implementation of a speedtest.net sensor."""
+
+    _attr_has_entity_name = True
+
+    entity_description: HuaweiSmartLogger3000EntityDescription
+
+    def __init__(
+        self,
+        coordinator: HuaweiSmartLogger3000DataCoordinator,
+        description: HuaweiSmartLogger3000EntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self.entity_id = ENTITY_ID_SENSOR_FORMAT.format(description.key)
+        self._attr_unique_id = f"huaweismartlogger3000_{description.key}"
+        self._state: StateType = None
+        self._attrs: dict[str, Any] = {}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.coordinator.api.HOST)},
+            entry_type=DeviceEntryType.SERVICE,
+            manufacturer="Huawei",
+            model="Smart Logger 3000",
+            configuration_url="https://support.huawei.com/enterprise/en/doc/EDOC1100108365/1a9e42de/webui-layout",
```

Comment:
The `configuration_url` creates a "Visit" link on the device page to make it easy for users to go to the devices' page, so it should be something like `f"https://{coordinator.api.host}"` or however the format of the URL to the device looks like.